### PR TITLE
REGISTRY-3289 fix, Allow anonymous users to access swagger UI extention

### PR DESCRIPTION
**Pull request by tmkasun**
In this PR:
- Wrap the `greg_swagger` page render login inside exec method
- Change registry access method as to access via `createTenantAwareAssetResources` using `tenantApi`
- Move separate `greg_swagger` extensions to store extensions directory and publisher extensions directory
- Remove es-extensions > commons directory since there are no common extensions kept in the location
- Add new logic to check for external swagger definition files referred in swagger < 2 version files to check for permission before trying to get the resource from the registry.
- Change `Show Swagger UI` button styling to match with the surrounding button group styling

### CODE REVIEW

File: modules/es-extensions/publisher/extensions/app/greg_swagger/pages/index.jag [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+        var value = '' + new Stream(new ByteArrayInputStream(content));
+        jsonValue = parse(value);
+        var url = config.server.https;
+
+        if (commonConstants.SWAGGER_MEDIA_TYPE.equals(resource.getMediaType())) {
+            /*log.info("wrong content type");
+             response.status = 400;
+             print("swagger UI generation is only avaialable for swagger artifacts," +
+             " Please specify a registry path of a swagger artifact!");
+             return;*/
```
Comment (daneshk) [inline modules/es-extensions/publisher/extensions/app/greg_swagger/pages/index.jag]: remove commented out codes

File: modules/es-extensions/store/extensions/app/greg_swagger/modules/utility.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+function isNotNullOrEmpty(checkText){
+
+    if(checkText == "" || checkText === 'null' || checkText == null || checkText.length <= 0){
+        return false;
+    }
+    return true;
+}
+
+function inputNotNullOrEmpty(jsonObj, paramNameArray){
```
Comment (daneshk) [inline modules/es-extensions/store/extensions/app/greg_swagger/modules/utility.js]: formatting issue. line 19, 21, 27

File: modules/es-extensions/store/extensions/app/greg_swagger/pages/index.jag [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+        var value = '' + new Stream(new ByteArrayInputStream(content));
+        jsonValue = parse(value);
+        var url = config.server.https;
+
+        if (commonConstants.SWAGGER_MEDIA_TYPE.equals(resource.getMediaType())) {
+            /*log.info("wrong content type");
+             response.status = 400;
+             print("swagger UI generation is only avaialable for swagger artifacts," +
+             " Please specify a registry path of a swagger artifact!");
+             return;*/
```
Comment (daneshk) [inline modules/es-extensions/store/extensions/app/greg_swagger/pages/index.jag]: remove commented out code

File: modules/es-extensions/publisher/extensions/app/greg_swagger/pages/index.jag (hunk 1 above)
Comment (tmkasun) [inline modules/es-extensions/publisher/extensions/app/greg_swagger/pages/index.jag]: Thanks, I have fixed it in [this commit](https://github.com/wso2/product-greg/pull/854/commits/93508144b9b01b4b424867b5632941c2166bf0e1)

File: modules/es-extensions/store/extensions/app/greg_swagger/pages/index.jag (hunk 3 above)
Comment (tmkasun) [inline modules/es-extensions/store/extensions/app/greg_swagger/pages/index.jag]: Thanks, I have fixed it in [this commit](https://github.com/wso2/product-greg/pull/854/commits/93508144b9b01b4b424867b5632941c2166bf0e1)

File: modules/es-extensions/store/extensions/app/greg_swagger/modules/utility.js (hunk 2 above)
Comment (tmkasun) [inline modules/es-extensions/store/extensions/app/greg_swagger/modules/utility.js]: Thanks, I have fixed it in [this commit](https://github.com/wso2/product-greg/pull/854/commits/93508144b9b01b4b424867b5632941c2166bf0e1)

File: modules/es-extensions/store/extensions/app/greg_swagger/modules/utility.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+    var input;
+    for (var i = 0; i < paramNameArray.length; i++) {
+        input = jsonObj[paramNameArray[i]];
+        if(input == "" || input === 'null' || input == null || input.length <= 0){
```
Comment (splinter) [inline modules/es-extensions/store/extensions/app/greg_swagger/modules/utility.js]: Formatting issue, line 32

File: modules/es-extensions/publisher/extensions/app/greg_swagger/pages/index.jag [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+           swaggerContent: <%=
+    jsonValue
+    %>,
+           swaggerChildren: <%=
+    childAPIs
+    %>,
+               url: url,
+               dom_id: "swagger-ui-container",
+               supportedSubmitMethods: ['get', 'post', 'put', 'delete', 'patch'],
+               onComplete: function(swaggerApi, swaggerUi){
```
Comment (tmkasun) [inline modules/es-extensions/publisher/extensions/app/greg_swagger/pages/index.jag]: Thanks @splinter for pointing out this formatting issue, I have fixed it in [this commit](https://github.com/wso2/product-greg/pull/854/commits/bfa37074378e690eb2cb40ecf3d5de50e684f9e3)

File: modules/es-extensions/publisher/extensions/app/greg_swagger/pages/index.jag [hunk 6]
```diff
@@ -0,0 +1,193 @@
...
+               },
+               onFailure: function(data) {
+                 log("Unable to Load SwaggerUI");
+               },
+               docExpansion: "none",
+               apisSorter: "alpha",
+               showRequestHeaders: false
+             });
+
+             function addApiKeyAuthorization(){
```
Comment (tmkasun) [inline modules/es-extensions/publisher/extensions/app/greg_swagger/pages/index.jag]: Thanks @splinter for pointing out this formatting issue, I have fixed it in [this commit](https://github.com/wso2/product-greg/pull/854/commits/bfa37074378e690eb2cb40ecf3d5de50e684f9e3)

File: modules/es-extensions/store/extensions/app/greg_swagger/app.js [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+app.server = function(ctx){
```
Comment (tmkasun) [inline modules/es-extensions/store/extensions/app/greg_swagger/app.js]: Thanks @splinter for pointing out this formatting issue, I have fixed it in [this commit](https://github.com/wso2/product-greg/pull/854/commits/bfa37074378e690eb2cb40ecf3d5de50e684f9e3)

File: modules/es-extensions/store/extensions/app/greg_swagger/modules/utility.js (hunk 4 above)
Comment (tmkasun) [inline modules/es-extensions/store/extensions/app/greg_swagger/modules/utility.js]: Thanks @splinter for pointing out this formatting issue, I have fixed it in [this commit](https://github.com/wso2/product-greg/pull/854/commits/bfa37074378e690eb2cb40ecf3d5de50e684f9e3)

**Pull request merged**